cloudca/resource_cloudca_static_nat.go

```diff
@@ -0,0 +1,85 @@
+package cloudca
+
+import (
+	"github.com/cloud-ca/go-cloudca/services/cloudca"
+	"github.com/hashicorp/terraform/helper/schema"
+	"fmt"
+)
+
+func resourceCloudcaStaticNat() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudcaStaticNatCreate,
+		Read:   resourceCloudcaStaticNatRead,
+		Delete: resourceCloudcaStaticNatDelete,
+
+		Schema: map[string]*schema.Schema{
+			"service_code": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A cloudca service code",
+			},
+			"environment_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of environment where tier should be created",
+			},
+			"organization_code": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Entry point of organization",
+			},
+
+			"public_ip_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The public IP to enable static NAT on",
+			},
+			"private_ip_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The private IP to enable static NAT on",
+			},
+		},
+	}
+}
+
+func resourceCloudcaStaticNatCreate(d *schema.ResourceData, meta interface{}) error {
+	resources := getResources(d, meta)
+	staticNatPublicIp := cloudca.PublicIp{
+		Id:          d.Get("public_ip_id").(string),
+		PrivateIpId: d.Get("private_ip_id").(string),
+	}
+	_, err := resources.PublicIps.EnableStaticNat(staticNatPublicIp)
+	if err != nil {
+		return fmt.Errorf("Error enabling static NAT: %s", err)
+	}
+	d.SetId(staticNatPublicIp.Id)
```

Comment:
Because static NAT isn't a "real" resource it's actually always non-`nil` - if it was `nil`, the call to `EnableStaticNat` would fail (because there is no public IP with id `nil`) and the function would have errored out.